BDArmory/Control/MissileFire.cs

```diff
@@ -2447,52 +2462,97 @@ IEnumerator GuardMissileRoutine(Vessel targetVessel, MissileBase ml)
                         }
                     case MissileBase.TargetingModes.Inertial:
                         {
-                            if (SetCargoBays())
+                            if (vesselRadarData) //no check for radar present, but off/out of juice
                             {
-                                yield return new WaitForSecondsFixed(2f);
-                                if (vessel == null || targetVessel == null) break;
-                            }
+                                float BayTriggerTime = -1;
+                                if (SetCargoBays())
+                                {
+                                    BayTriggerTime = Time.time;
+                                    //yield return new WaitForSecondsFixed(2f);
+                                    if (vessel == null || targetVessel == null) break;
+                                }
 
-                            float attemptStartTime = Time.time;
-                            MissileLauncher mlauncher = ml as MissileLauncher;
-                            if (mlauncher)
-                            {
-                                float angle = 999;
-                                float turretStartTime = attemptStartTime;
-                                if (mlauncher.missileTurret)
+                                float attemptStartTime = Time.time;
+
+                                float attemptLockTime = Time.time;
+                                while (ml && vesselRadarData && (!vesselRadarData.locked || (vesselRadarData.lockedTargetData.vessel != targetVessel)) && Time.time - attemptLockTime < 2f)
                                 {
-                                    while (Time.time - turretStartTime < Mathf.Max(targetScanInterval / 2f, 2) && mlauncher && targetVessel && angle > mlauncher.missileTurret.fireFOV)
+                                    if (vesselRadarData.locked)
```

Comment:
Part of the point/implementation of INS was that is just needed TWS to launch against a target; this should instead be something like 
```
if (vesselRadarData)
{
    INSTarget = vesselRadarData.detectedRadarTarget(targetVessel, this);
    if (!INSTarget.exists && irsts.Count > 0 && _irstsEnabled)
    {
       INSTarget = vesselRadarData.activeIRTarget(null, this);
    }
    if (vessel && INSTarget.exists && targetVessel)
    {
        Vector3 TargetLead = MissileGuidance.GetAirToAirFireSolution(ml, INSTarget.vessel.CoM,  INSTarget.vessel.Velocity());
        designatedINSCoords = VectorUtils.WorldPositionToGeoCoords(TargetLead, INSTarget.vessel.mainBody);
     }...
```